svc/pkg/mm/worker/redis-scripts/lobby_find.lua

```diff
@@ -64,16 +70,36 @@ elseif query.kind.lobby_group ~= nil then
 		-- instead of `max_players_party` because `max_players_party` is
 		-- only relevant when joining a lobby with the exact lobby ID (i.e.
 		-- where `max_players_direct` would normally be used).
-		local lobby = redis.call('ZRANGEBYSCORE', key_available_spots, table.getn(query.players), '+inf', 'WITHSCORES',
+		local lobby = redis.call('ZRANGEBYSCORE', key_available_spots, #query.players, '+inf', 'WITHSCORES',
 			'LIMIT', '0', '1')
-		if table.getn(lobby) > 0 then
+		if #lobby > 0 then
 			local lobby_id = lobby[1]
 			local available_spots = tonumber(lobby[2])
+			local correct_tags = 0
+
+			-- Verify tags
+			if #lobby_tag_names > 0 then
+				-- Fetch specified lobby tags
+				local key_lobby_tags = '{global}:mm:lobby:' .. lobby_id .. ':tags'
+				local lobby_tags = redis.call('HMGET', key_lobby_tags, unpack(lobby_tag_names))
+				
+				-- Check that all tags are correct
+				for tag_idx, tag in ipairs(lobby_tags) do
+					if query.tags[lobby_tag_names[tag_idx]] ~= tag then
+						break
+					end
+
+					correct_tags = correct_tags + 1
+				end
+			end
 
-			-- Set as best lobby if no lobby selected or lobby is more full
-			if best_lobby_id == nil or available_spots < best_available_spots then
-				best_lobby_id = lobby_id
-				best_available_spots = available_spots
+			-- Check if the correct number of tags have been parsed
+			if correct_tags == #lobby_tag_names then
```

Comment:
Could this if statement be combined with the inner one?

Reply:
https://github.com/rivet-gg/rivet/pull/197/commits/4d08c9ca1fe72741435e436e5e7aa8426c25943d